lib/units/test/unit_test.cpp

```diff
@@ -31,6 +31,15 @@ TEST(UnitTest, construct_bad_string) {
   EXPECT_THROW(Unit("abcde"), except::UnitError);
 }
 
+TEST(UnitTest, custom_unit_strings_get_rejected) {
+  // Custom (counting) units and equation units are rejected.
+  for (const auto &str :
+       {"CXUN[0]", "CXUN[51]", "CXUN{1023]", "CXCUN[0]", "CXCUN[15]",
```

Comment:
Is the `{1023]` intentionally using mixed bracket types?

Reply:
No, it's a typo